substrate/client/consensus/beefy/src/communication/gossip.rs

```diff
@@ -495,6 +497,95 @@ pub(crate) mod tests {
 	};
 	use sp_keystore::{testing::MemoryKeystore, Keystore};
 
+	pub(crate) struct TestNetwork {
+		report_sender: futures::channel::mpsc::UnboundedSender<PeerReport>,
+	}
+
+	impl TestNetwork {
+		pub fn new() -> (Self, futures::channel::mpsc::UnboundedReceiver<PeerReport>) {
+			let (tx, rx) = futures::channel::mpsc::unbounded();
+
+			(Self { report_sender: tx }, rx)
+		}
+	}
+
+	impl NetworkPeers for TestNetwork {
+		fn set_authorized_peers(&self, _: std::collections::HashSet<PeerId>) {
+			todo!()
```

Comment:
nit: what about `unreachable!()`?

Reply:
Rust analyzer filed them out automatically. In the end it is a panic :D

Reply:
yeah. Functionally is the same soup, but maybe not the semantic. I  was suggesting it just because I'd find `todo!` macro to be placed where we intend to eventually add something in the future.

Reply:
Because it is you ser :) 

https://github.com/paritytech/polkadot-sdk/pull/4015/commits/c5f616b97e65a20ab72c1f66f584de8080e10a62